src/lsp-server.ts

```diff
@@ -983,4 +984,47 @@ export class LspServer {
         }
         return callsResult;
     }
+
+    async inlayHints(
+        params: lspinlayHints.InlayHintsParams
+    ): Promise<lspinlayHints.InlayHintsResult> {
+        const file = uriToPath(params.textDocument.uri);
+        this.logger.log('inlayHints', params, file);
+        if (!file) {
+            return { inlayHints: [] };
+        }
+
+        const doc = this.documents.get(file);
+        if (!doc) {
+            return { inlayHints: [] };
+        }
+
+        const start = doc.offsetAt(params.range?.start ?? {
+            line: 0,
+            character: 0
+        });
+        const end = doc.offsetAt(params.range?.end ?? {
+            line: doc.lineCount + 1,
+            character: 0
+        });
+        const result = await this.tspClient.request(
+            CommandTypes.ProvideInlayHints,
+            {
+                file,
+                start: start,
+                length: end - start
+            }
+        );
+
```

Comment:
I think we have to cater for the case when the tsserver doesn't support that command since this language server can resolve typescript from the project and there is no guarantee that this will be supported. We could potentially just return an empty result in that case.